analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Invocation.cs

```diff
@@ -59,44 +60,83 @@ protected override ProgramState[] Process(SymbolicContext context, IInvocationOp
             _ when invocation.TargetMethod.Name == nameof(object.Equals) => ProcessEquals(context, invocation),
             _ when invocation.TargetMethod.IsAny(KnownType.System_String, nameof(string.IsNullOrEmpty), nameof(string.IsNullOrWhiteSpace)) =>
                 ProcessIsNotNullWhen(state, invocation.WrappedOperation, invocation.Arguments[0].ToArgument(), false),
-            _ => ProcessIsNotNullWhen(state, invocation),
+            _ => ProcessArgumentAttributes(state, invocation),
         };
     }
 
-    private static ProgramState[] ProcessIsNotNullWhen(ProgramState state, IInvocationOperationWrapper invocation)
+    private static ProgramState[] ProcessArgumentAttributes(ProgramState state, IInvocationOperationWrapper invocation)
     {
-        foreach (var argument in invocation.Arguments.Select(x => x.ToArgument())) // TODO: support attributes on more than one argument
+        IReadOnlyCollection<ProgramState> states = new[] { state };
+        foreach (var argument in invocation.Arguments.Select(x => x.ToArgument()).Where(x => x.Parameter is not null))
         {
-            if (argument.Parameter?.GetAttributes().FirstOrDefault(x => x.HasName("NotNullWhenAttribute")) is { } attribute
-                && attribute.TryGetAttributeValue<bool>("returnValue", out var returnValue))
+            foreach (var attribute in argument.Parameter.GetAttributes())
             {
-                return ProcessIsNotNullWhen(state, invocation.WrappedOperation, argument, returnValue);
+                states = ProcessArgumentAttribute(states, invocation, argument, attribute);
             }
         }
-        return new[] { state };
+        return states.ToArray();
+    }
+
+    private static IReadOnlyCollection<ProgramState> ProcessArgumentAttribute(IEnumerable<ProgramState> states,
+                                                                              IInvocationOperationWrapper invocation,
+                                                                              IArgumentOperationWrapper argument,
+                                                                              AttributeData attribute)
+    {
+        var ret = new List<ProgramState>();
```

Comment:
The `new List<ProgramState>();` should be avoided. The `DefineConstraintsFromKnownResult` approach should make this temporary list obsolete as we can pass the state from argument to argument and adding the bool constraint to the invocation operation helps the following arguments to plugin at the right places and do their explosions.

Reply:
Argument 1 can produce a collection of states. Each of them needs to be processed against Argument 2 independently.

Reply:
I tried the argument processor before, it didn't work well. 
New idea (unexplored, for another PR, made a note to myself): Maybe in Invocation, we can "do nothing" when the invocation already has any bool constraint. And explode for True/NotNull and False/NotNull on the "normal" handling.

Reply:
>The attributes are queried in each state iteration but return the same values every time

What line does it refer to? 72? They are queried for different parameters.

Reply:
..got it, attribute values, not attributes themselves, right?